conventional_pre_commit/hook.py

```diff
@@ -47,6 +50,10 @@ def main(argv=[]):
         )
         return RESULT_FAIL
 
+    if not args.strict:
+        if format.has_autosquash_prefix(message):
+            return RESULT_SUCCESS
```

Comment:
Hmm, do we we need to entirely shortcut the conventional commits check?

I was thinking we would just allow the addition of the prefix, but the rest of the message still has to be in conventional commit format. Otherwise this lets passes anything starting with `fixup!` etc. and kind of defeats the purpose of this hook?

Reply:
That was my first thought as well. But then I started thinking about usage scenarios. It turns out that fixup-like commits can only be based on some other commits. A user must specify the SHA1 of a commit that should be fixed. If the hook is active it's impossible to make a regular commit ill-formed in terms of Conventional Commits formatting. The same thing for doing rebase with auto squashing: if the result commit message doesn't match, it's rejected. 
There is only one corner case, I see here when a user intentionally makes a commit that starts with "fixup! " (or similar). Which I agree could happen, but it doesn't make any sense in terms of the workflows where fixup-like commits are used.

Based on this logic, I decided to check a prefix as earlier as possible and return fast. Another thing is in the code. It's possible to extend `format.is_conventional` with one more regexp for checking prefixes, but it might be a little bit cumbersome and doesn't fully match the name of the function.

On the other hand, I don't mind implementing checking the rest of the message. I just feel it might be redundant and overcomplicated.

What do you think?

Reply:
> A user must specify the SHA1 of a commit that should be fixed. If the hook is active it's impossible to make a regular commit ill-formed in terms of Conventional Commits formatting. The same thing for doing rebase with auto squashing

Fair enough, that's a good point.